packages/react-devtools-shared/src/__tests__/inspectedElementContext-test.js

```diff
@@ -1805,4 +1805,136 @@ describe('InspectedElementContext', () => {
 
     done();
   });
+
+  describe('inline errors and warnings', () => {
+    async function getErrorsAndWarningsForElementAtIndex(index) {
+      const id = ((store.getElementIDAtIndex(index): any): number);
+
+      let errors = null;
+      let warnings = null;
+
+      function Suspender({target}) {
+        const {getInspectedElement} = React.useContext(InspectedElementContext);
+        const inspectedElement = getInspectedElement(id);
+        errors = inspectedElement.errors;
+        warnings = inspectedElement.warnings;
+        return null;
+      }
+
+      await utils.actAsync(
+        () =>
+          TestRenderer.create(
+            <Contexts
+              defaultSelectedElementID={id}
+              defaultSelectedElementIndex={index}>
+              <React.Suspense fallback={null}>
+                <Suspender target={id} />
+              </React.Suspense>
+            </Contexts>,
+          ),
+        false,
+      );
+
+      return {errors, warnings};
+    }
+
+    it('during render get recorded', async done => {
+      const Example = () => {
+        console.error('render error');
+        console.warn('render warning');
+        return null;
+      };
+
+      const container = document.createElement('div');
+      await utils.actAsync(() =>
+        ReactDOM.render(<Example repeatWarningCount={1} />, container),
+      );
+
+      const data = await getErrorsAndWarningsForElementAtIndex(0);
+      expect(data).toMatchSnapshot();
+
+      done();
+    });
+
+    it('during render get deduped', async done => {
+      const Example = () => {
+        console.error('render error');
+        console.error('render error');
+        console.warn('render warning');
+        console.warn('render warning');
+        console.warn('render warning');
+        return null;
+      };
+
+      const container = document.createElement('div');
+      await utils.actAsync(() =>
+        ReactDOM.render(<Example repeatWarningCount={1} />, container),
+      );
+
+      const data = await getErrorsAndWarningsForElementAtIndex(0);
+      expect(data).toMatchSnapshot();
```

Comment:
What do you think about using inline snapshots here instead? They seem small enough that they don't drown out the test. Personally, I prefer them since I can understand the expected behavior by viewing a single file.

Reply:
I'm game!

Reply:
Weird. I tried adding one to this test and got a Jest error:
> Couldn't locate all inline snapshots

Reply:
Seems to be caused by using the callback style for tests. Or maybe mixing it with `async`. Since we don't need the callback style I went ahead and removed it. Maybe jest used to require and explicit `done()` in async tests?